File: week5/ex3_taeho.js
function solution(sequence, k) {
    
    const partialSumArray = [0]; 

    let partialSum = 0;
    for (let num of sequence) {
        partialSum += num;
        partialSumArray.push(partialSum);
    }

    let answer = [0 , sequence.length+1]
    let left = 0 ,right = 1;
    while (right < partialSumArray.length) {
        let sumOfInterval = partialSumArray[right] - partialSumArray[left];  
        if (sumOfInterval === k) {
            if (right - left - 1 < answer[1] - answer[0]){
                answer = [left,right-1];
            }
        left++;
        } else if(sumOfInterval > k){ 
        left++;
        } else {
        right++;
        }
    }
    return answer;
}

File: week14/ex3_taeho.py
def solution(x, y, n):

    dp = {}
    dp[x] = 0
    
    for num in range(y+1):
        ls = []
        if num - n in dp:
            ls.append(dp[num-n]+1)
        if num%2 == 0 and num/2 in dp:
            ls.append(dp[num/2] + 1)
        if num%3 == 0 and num/3 in dp:
            ls.append(dp[num/3] + 1)
        if ls:
            dp[num] = min(ls)
            
    return dp[y] if y in dp else -1
    